runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -44,7 +46,7 @@
  *  stream.
  *  TODO: Generalize to all SMR objects.
```

Comment:
CheckpointWriter is generalize to Maps but not to all SMR objects. The implementation is heavily relying on maps API.